inc/question.class.php

```diff
@@ -456,7 +456,7 @@ public function prepareInputForAdd($input) {
    public function prepareInputForUpdate($input) {
       // global $DB;
 
-      if (!in_array($input['fieldtype'], ['checkboxes', 'multiselect'])) {
+      if (isset($input['fieldtype']) && isset($input['default_values']) && !in_array($input['fieldtype'], ['checkboxes', 'multiselect'])) {
```

Comment:
here, if ```$input['fieldtype']``` is not set, you have to read ```$this->fields['fieldtype']```. If not, you may do your str_replace on the types you want to exclude.

Also if @stonebuzz is right, it would be safer to check if the type is ```textarea``` instead of checking it is not one of some types.